File: src/main/java/com/skyspace999/service/CancellationService.java
package com.skyspace999.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspace999.domain.Cancellation;
import com.skyspace999.dto.CancellationDTO;
import com.skyspace999.dto.CancellationSearchDTO;
import com.skyspace999.dto.CancellationPageDTO;
import com.skyspace999.dto.CancellationConvertCriteriaDTO;
import com.skyspace999.service.GenericService;
import com.skyspace999.dto.common.RequestDTO;
import com.skyspace999.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CancellationService extends GenericService<Cancellation, Integer> {

	List<Cancellation> findAll();

	ResultDTO addCancellation(CancellationDTO cancellationDTO, RequestDTO requestDTO);

	ResultDTO updateCancellation(CancellationDTO cancellationDTO, RequestDTO requestDTO);

    Page<Cancellation> getAllCancellations(Pageable pageable);

    Page<Cancellation> getAllCancellations(Specification<Cancellation> spec, Pageable pageable);

	ResponseEntity<CancellationPageDTO> getCancellations(CancellationSearchDTO cancellationSearchDTO);
	
	List<CancellationDTO> convertCancellationsToCancellationDTOs(List<Cancellation> cancellations, CancellationConvertCriteriaDTO convertCriteria);

	CancellationDTO getCancellationDTOById(Integer cancellationId);







}

File: skyspace999_ui/src/services/CancellationService.js
import http from "../http-common"; 

class CancellationService {
  getAllCancellations(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/cancellation/cancellations`, searchDTO);
  }

  get(cancellationId) {
    return this.getRequest(`/cancellation/${cancellationId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/cancellation?field=${matchData}`, null);
  }

  addCancellation(data) {
    return http.post("/cancellation/addCancellation", data);
  }

  update(data) {
  	return http.post("/cancellation/updateCancellation", data);
  }
  
  uploadImage(data,cancellationId) {
  	return http.postForm("/cancellation/uploadImage/"+cancellationId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CancellationService();
